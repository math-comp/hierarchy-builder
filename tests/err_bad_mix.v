From HB Require Import structures.

Module Test.
  HB.mixin Record Mixin T := {
    zero: T;
  }.

  HB.structure Definition Struct := { T of Mixin T }.

  HB.instance Definition struct_bool := Mixin.Build bool true.

  Module Exports.
    HB.reexport.
  End Exports.
End Test.
(** Uncommenting any of these two prevents the issue *)
(* Export Test.Exports. *)
(* HB.export Test. *)

Fail HB.instance Definition struct_nat := Test.Mixin.Build nat 0.
