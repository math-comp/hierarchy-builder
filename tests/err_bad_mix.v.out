Interactive Module Test started
Interactive Module Exports started
The command has indeed failed with message:
HB: Test.Mixin.axioms_ 
is not a factory or its library was not correctly imported
